metpy/calc/indices.py

```diff
@@ -51,3 +52,51 @@ def precipitable_water(dewpt, p, top=400 * units('hPa')):
     pw = -1. * (np.trapz(w.magnitude, pres_layer.magnitude) * (w.units * pres_layer.units) /
                 (g * rho_l))
     return pw.to('millimeters')
+
+
+@exporter.export
+@check_units('[speed]', '[speed]', '[pressure]', '[length]')
+def mean_wind_pressure_weighted(u, v, p, hgt, depth, bottom=None, obs_only=False):
+    r"""Calculate pressure-weighted mean wind through a layer.
+
+    Layer bottom and depth specified in meters AGL.
+
+    Parameters
+    ----------
+    u : array-like
+        U-component of wind.
+    v : array-like
+        V-component of wind.
+    p : array-like
+        Atmospheric pressure profile
+    hgt : array-like
+        Heights from sounding
+    depth: `pint.Quantity`
+        The depth of the layer in meters.
+    bottom: `pint.Quantity`
+        The bottom of the layer in meters AGL.
+        Default is the first observation, assumed to be the surface.
+
+    Returns
+    -------
+    `pint.Quantity`
+        u_mean: u-component of layer mean wind, in m/s
+    `pint.Quantity`
+        v_mean: v-component of layer mean wind, in m/s
+
+    """
+    u = u.to('meters/second')
+    v = v.to('meters/second')
+    if bottom:
+        bottom = bottom + hgt[0]
+    w_int = get_layer(p, u, v, heights=hgt, bottom=bottom, depth=depth)
```

Comment:
Instead of using indexing, it's more readable to unpack into meaningful variable names. Something like:

`layer_p, layer_u, layer_v = get_layer(...)`